src/dmd/mtype.d

```diff
@@ -6183,6 +6183,27 @@ extern (C++) final class TypeNull : Type
     }
 }
 
+/***********************************************************
+ * Encapsulate Parameters* so .length and [i] can be used on it.
+ */
+struct ParameterList
+{
+    Parameters* parameters;
+
+    size_t length()
+    {
+        return Parameter.dim(parameters);
+    }
+
+    Parameter opIndex(size_t i)
```

Comment:
`ref opIndex`?

Reply:
A `Parameter` is a class, and is already by ref.